### DISCUSSION TITLE
Consider consolidating Discord discussions onto GitHub

### DISCUSSION THREAD

**Discussion by vladh** [jj-vcs/jj#4843] [Ideas]
Hi! 👋🏻  Would it make sense to consolidate all discussions on GitHub, instead of having them on GitHub and Discord? I'm referring specifically to the Discord “questions” section, which seems to be very similar to the GitHub “discussions”. I'm not referring to ephemeral chat.

It's a bit cumbersome, counterintuitive and duplication-prone to have to search both Discord and GitHub to find whether a question has been asked before. And GitHub has the advantage of being a publicly accessible archive. Discord might work fine for ephemeral chat, and it is trivial to migrate ephemeral chat away from Discord later. But this isn't true for discussions — locking valuable community discussions into a platform that isn't publicly viewable, and is difficult to migrate away from, leads to information being more difficult to access now, and might lead to it being lost later.

Happy to hear what everyone in the community thinks! Perhaps there are better platforms that GitHub too.

**Comment by tim-janik** [comment]
I second that and might add, to prepare in case either platform closes up, blocks a project or sanctions a region:
- Github discussions and issues can be properly archived via [git-bug](https://github.com/git-bug/git-bug) (and the wiki can be cloned as a git repo).
- For discord, while there are scripts available that try to download conversations, [account automation is against the terms of service](https://www.downloadsource.net/how-to-download-discord-channel-and-direct-message-dm-history/n/14222/#1) so the attempt might get you blocked.

**Comment by neongreen** [comment]
I, for one, would simply not have the discussions on GitHub that I’m having on Discord.

Whenever I have an obscure issue with an OSS project, I expect that if the answer exists _at all_, it will be either in GitHub issues, or Discord. And the Discord one is often more relevant.

If I want to know something informal (what does the author of X think about competitor Y; is feature Z planned; etc) — then it’s usually only on Discord.

For projects that don’t have Discord, I don’t see the same kind of content being available elsewhere — I just fail to find it at all. Searching Discord, to me, is like appending “site:reddit.com” in google.

I don’t know if others have had the same experience, and maybe it’s limited to the kinds of projects I engage with, but I would be rather sad if Discord was phased out in favor of GitHub discussions, Discourse, etc. 

I would be happy though if smth like Discourse existed in parallel. I’ve noticed that forums also spawn some kinds of content that chats don’t — high-effort posts, scripts/wrappers/tools, mini-tutorials, collaborative searches for workarounds, lists of customizations/plugins.

To sum up — I’d like a forum to exist (a better one than GitHub discussions), but not as a discord replacement. I think the value of having a forum and a chat is higher than the value of having a single place to search. 

(And also discord is a really good chat. But that’s more subjective of course.)

**Reply by neongreen** [reply]
Update: OP clarified its about the “questions” Discord channel only. I don’t have an opinion on the questions channel, but I suspect people who use Discord might just switch to asking questions in #general then, so it may not accomplish much. But idk.

**Reply by emilazy** [reply]
I agree that we definitely need a real‐time chat medium, even for support and development discussions, though I think it doesn’t necessarily have to be Discord.

**Reply by mcepl** [reply]
> Whenever I have an obscure issue with an OSS project, I expect that if the answer exists _at all_, it will be either in GitHub issues, or Discord. And the Discord one is often more relevant.

I just cannot express how incredibly sad and hopeless about the humanity such statements make me. https://xkcd.com/743/

**Comment by vladh** [comment]
Sorry, just to be clear, my question is about merging the Discord “questions” section into the GitHub “discussions” section, which both seem to be used in the same way.

I think that might not have been clear, I'm editing the OP to clarify.

Of course, ephemeral chat is different, and can continue to be maintained either on Discord or wherever.

**Comment by martinvonz** [comment]
Sounds like a good idea to me! I think we can at least add a note in the #questions channel on Discord saying that we prefer if they ask on GitHub instead. I'll check what people on Discord think about that.

**Comment by ddevault** [comment]
(Some of) the Discord server has been bridged to IRC, which is a big improvement, though IRC has some usability issues so there is a "midding middle" which might be better served by something like Matrix or Zulip.
